src/omnicore/omnicore.cpp

```diff
@@ -551,6 +551,38 @@ int mastercore::GetEncodingClass(const CTransaction& tx, int nBlock)
     bool hasOpReturn = false;
     bool hasMoney = false;
 
+    /* Fast Search
```

Comment:
Heh of course :)  This was one of the substantial speed ups, and I'd be keen to leave it in.

You are quite correct, it fractionally slows down parsing Omni transactions.  However it also fractionally *speeds up* the rate at which we can identify and drop non-Omni transactions.

For example sake let's say there are 2000 BTC transactions and 5 Omni transactions in a particular block.  We're going to do "something" 2000 times and "something else" 5 times.  It makes sense to prioritize doing the "something" faster :)